### ISSUE TITLE
Handle unknown unitary operations in quirk export

### ISSUE THREAD

**Issue by balopat** [quantumlib/Cirq#4349]
**Is your feature request related to a use case or problem? Please describe.**

I was trying to export a circuit that was compiled to a sqrtISWAP gateset to inspect in quirk for a presentation. However the export failed.

```
TypeError: Unrecognized operation: (cirq.ISWAP**-0.5).on(cirq.LineQubit(3), cirq.LineQubit(4))
```

**Describe the solution you'd like**

It would be nice if the quirk exporter could handle unknown gates automatically, quirk does have the ability to create custom gates from unitaries.

**[optional] Describe alternatives/workarounds you've considered**

Use the `prefer_unknown_gate_to_faliure=True` results in too many of these gates, but at least the URL is generated: 

![image](https://user-images.githubusercontent.com/635073/126665959-669e6d4f-815d-4873-a77d-52f07aa9ce08.png)

Maybe I could do the custom gate creation and replacement in the URL manually, but I probably won't :)

**What is the urgency from your perspective for this issue? Is it blocking important work?**

P3 - I'm not really blocked by it, it is an idea I'd like to discuss / suggestion based on principle

**Comment by mhucka** [comment]
I'm closing this as a nice-to-have-but-we-don't-have-time feature request. If this is incorrect or something changes, someone please feel free to reopen it.
